accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,632 @@
+
+Web Public Key Infrastructure ([Web PKI](https://en.wikipedia.org/wiki/Public_key_infrastructure)) Certificate Authorities (CA) currently use a similar approach when they publish all issued certificates to a [transparency log](https://certificate.transparency.dev/).
+
+### CI/CD OIDC provider support
+Today only GitHub Actions is fully supported by Fulcio. We’d like to see support added for any public CI/CD service that can meet these requirements:
+- OIDC ID tokens identifying the current workflow/run/build.
+- ID token with a custom audience (needs to be set to `sigstore` for Fulcio to accept it).
+- Claims about the code repository, commit, build and actor (see a detailed table of required ID token claims below)
+
+As of today, Google Cloud Build supports custom claims but neither GitLab Pipelines nor CircleCI offers this. [GitLab](https://gitlab.com/gitlab-org/gitlab/-/issues/356986) is planning to add support for a custom audience.
+
+The plan is to start with direct vendor support in Fulcio. Fulcio will need to be patched to produce certificates containing custom claims data from Circle CI, GitLab and Google Cloud Build.
+
+We will also work with OpenSSF and other package ecosystems adopting [Sigstore](https://www.sigstore.dev/) on an official claims set that any provider can implement. Once we have this each vendor could be added to Fulcio by simply specifying the standard OIDC client configuration.
+
+#### Comparison of OIDC ID token support across CI/CD systems
+
+| Name               | OIDC ID token | Custom audience | Comment                                                                                                                                                                |
+|--------------------|--------------|-----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| GitHub Actions     | yes          | yes             | Uses “workflow run” as the subject                                                                                                 |
+| GitLab Pipelines   | yes          | no              | Subject is the project path. See [documentation](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html).                                                                                                                   |
+| CircleCI           | yes          | no              | Uses [workload ID](https://circleci.com/docs/2.0/openid-connect-tokens/) that includes a notion of “user ID” in the Fsubject claim. There is a [feature request to make this configurable](https://circleci.canny.io/cloud-feature-requests/p/customizable-audience-claim-in-oidc-tokens) |
+| Google Cloud Build | yes          | yes             |                                                                                                                                                     |
```

Comment:
Cirrus CI also supports OIDC tokens and now allows to override the audience (see https://github.com/cirruslabs/cirrus-ci-docs/pull/1058). Will appreciate if Cirrus CI can be also included.  We also support a [pretty similar to GHA set of claims](https://oidc.cirrus-ci.com/.well-known/openid-configuration).

```suggestion
| Google Cloud Build | yes          | yes             |                                                                                                                                                     |
| Cirrus CI.                  | yes          | yes             |                                                                                                                                                     |
```

Reply:
@fkorotkov love it! The next step would be adding support for it in Fulcio: https://github.com/sigstore/fulciox - I think some groundwork might need to be done here to extend the signing certificate. Here's an open issue for adding GitLab support for reference: https://github.com/sigstore/fulcio/issues/243